app/views/contexts/summary/calculated_summary_block.py

```diff
@@ -30,20 +31,34 @@ def __init__(
         location: LocationType,
         return_to: str | None,
         return_to_block_id: str | None = None,
+        return_to_list_name: str | None = None,
+        return_to_list_item_id: str | None = None,
         progress_store: ProgressStore,
         routing_path_block_ids: Iterable[str],
         supplementary_data_store: SupplementaryDataStore,
     ) -> None:
         """
         A Calculated summary block that is rendered as part of a grand calculated summary
+
+        If the GCS is in a repeating section, and the calculated summary also is
+        then the list name and item id need to be used to build the calculated summary change link
+        but if the GCS is repeating and the CS is not, the list parameters in the change link should be set to None
         """
 
         self.id = block_schema["id"]
         self.title = block_schema["calculation"]["title"]
         self._return_to = return_to
         self._return_to_block_id = return_to_block_id
+        self._return_to_list_name = return_to_list_name
+        self._return_to_list_item_id = return_to_list_item_id
         self._block_schema = block_schema
         self._schema = schema
+        if location and self._schema.is_block_in_repeating_section(self.id):
```

Comment:
Do we need the check for location here? Can it be None?

Reply:
good spot - it can't, removing 👍